src/app/components/score-chart/score-chart.component.ts

```diff
@@ -1,19 +1,64 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, Input, OnChanges, OnInit, SimpleChanges } from '@angular/core';
 import * as Highcharts from 'highcharts';
+import more from 'highcharts/highcharts-more';
+import { parseISO } from 'date-fns';
+import { mean } from 'lodash';
+
+more(Highcharts);
 
 @Component({
   selector: 'app-score-chart',
   templateUrl: './score-chart.component.html',
   styleUrls: ['./score-chart.component.sass']
 })
-export class ScoreChartComponent implements OnInit {
+export class ScoreChartComponent implements OnChanges {
 
-  private options: any = {};
+  @Input()
+  public historicalData: string[];
 
-  constructor() { }
+  @Input()
+  public label: string;
 
-  ngOnInit(): void {
-    Highcharts.chart('container', this.options);
-  }
+  @Input()
+  public currentValue = '- -';
+
+  public averageValue = '--';
+  public Highcharts: typeof Highcharts = Highcharts;
+  public chartOptions: Highcharts.Options = {
+    title: {
+      text: 'Track Your Progress'
+    },
+
+    yAxis: {
+        title: {
+            text: 'Score'
+        }
+    },
 
+    xAxis: {
+        accessibility: {
+            rangeDescription: 'Date'
+        }
+    },
+
+    plotOptions: {
+        series: {
+            label: {
+                connectorAllowed: false
+            }
+        }
+    },
+    series: [],
+  };
+
+  ngOnChanges(changes: SimpleChanges): void {
```

Comment:
I don't think we need this lifecycle hook:

1. `averageValue` can be a computed property (which will update itself when `historicalData` changes):
```typescript
get averageValue(): number {
  return mean(/* ... */)
}
```
2. `chartOptions` could be a property:
```typescript
get chartOptions() {
  return {
    /*...*/,
    series: [{
      /*...*/,
      data: this.historicalData./*...*/
    }]
  };
}
```

Reply:
That's true. I am just used to using lifecycle hook for variable changes in general. Feel like it is one single place for all the updates. but using get is more neat

Reply:
@ansonmiu0214 I tried using get with Highcharts options data, it seems like Highcharts will keep calling the get function and we end up calculating the data every time the get is called. This makes the rendering super slow and disable the tooltip function as well